### PR TITLE
add map int64

### PR DESCRIPTION
**Pull request by duguhaotian**
1. fix lose ) and \n
2. support map string int64

Signed-off-by: duguhaotian <knowledgehao@163.com>

### CODE REVIEW

**Comment by duguhaotian** [comment]
@giuseppe please check 🙂

File: src/common_c.py [hunk 1]
```diff
@@ -1183,6 +1208,136 @@ def generate_json_common_c(out):
...
+        size_t len = YAJL_GET_OBJECT (src)->len;
+        ret = safe_malloc (sizeof(*ret));
+        ret->len = len;
+        ret->keys = safe_malloc ((len + 1) * sizeof (char *));
+        ret->values = safe_malloc ((len + 1) * sizeof (int64_t));
+        for (i = 0; i < len; i++)
+        {
+            const char *srckey = YAJL_GET_OBJECT (src)->keys[i];
+            yajl_val srcval = YAJL_GET_OBJECT (src)->values[i];
+            ret->keys[i] = strdup (srckey ? srckey : "");
```
Comment (giuseppe) [inline src/common_c.py:1281]: should this be `safe_strdup`?

File: src/common_c.py [hunk 2]
```diff
@@ -213,6 +213,31 @@ def generate_json_common_c(out):
...
 }
 
+char *
+safe_strdup (const char *src)
+{
+    char *dst = NULL;
+
+    if (src == NULL)
+        return NULL;
+    dst = strdup(src);
```
Comment (giuseppe) [inline src/common_c.py:223]: please add a space between `strdup` and `(`

File: src/common_c.py [hunk 3]
```diff
@@ -213,6 +213,31 @@ def generate_json_common_c(out):
...
+    if (dst == NULL)
+        abort();
```
Comment (giuseppe) [inline src/common_c.py:225]: same here for abort

**Comment by duguhaotian** [comment]
@giuseppe fixed

**Review by giuseppe — APPROVED** [review]
LGTM

**Pull request merged**